### PR TITLE
feat: pass a contract address to verifyCiphertext

### PR DESCRIPTION
**Pull request by dartdart26**
Will be needed in the future to verify the input ZKPoK.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -116,18 +116,14 @@ import "fhevm/lib/TFHE.sol";
...
-  function add(bytes calldata encryptedValue) public {
-    euint32 value = TFHE.asEuint32(encryptedValue);
-    counter = counter + value;
-  }
-
-  function getCounter(bytes32 publicKey) returns (bytes memory) {
-    return TFHE.reencrypt(counter, publicKey);
+  function add(einput valueInput, bytes calldata inputProof) public {
+    euint32 value = TFHE.asEuint32(valueInput, inputProof);
+    counter = TFHE.add(counter, value);
```
Comment (jatZama) [inline README.md:121]: Please add TFHE.allow(counter, address(this)); here!

**Review by jatZama — APPROVED** [review]
I just added a minor comment concerning the README example, otherwise all good.

File: README.md (hunk 1 above)
Comment (dartdart26) [reply README.md:121]: Can you explain the reasoning?

Comment (jatZama) [reply README.md:121]: because counter is a state variable, so if you want to persist it in ACL after the transaction it is mandatory.

Comment (dartdart26) [reply README.md:121]: Discussed offline - obviously needed if further operations are to be done on the counter. Adding the allow() call.

**Review by jatZama — APPROVED** [review]

**Pull request merged**